### ISSUE TITLE
Enhancement: Normalise gradient accumulation for high fan-out neurons

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1872]
## Context

Back propagation isn't improving large complex creature scores as expected (#1869). Analysis of `TopologicalBackpropagation.ts` reveals that error signals from multiple downstream paths are **summed without normalisation** (line 92). The `targetDeltaCount` is tracked (line 276) but never used for averaging.

In NEAT topologies, neurons have wildly different fan-out — some connect to 50+ downstream neurons while others connect to just 1-2. This creates gradient magnification: high fan-out neurons receive disproportionately large accumulated errors, leading to oversized weight/bias updates that can destabilise learning in complex networks.

## What needs to be done

Introduce optional gradient normalisation in the topological backward pass to balance gradient magnitude across neurons with different connectivity:

1. **Average vs sum strategy** — In `TopologicalBackpropagation.ts`, when computing `totalDelta` (line 92), provide an option to normalise by `targetDeltaCount`:
   ```typescript
   const totalDelta = config.normaliseGradients
     ? targetDeltaSum[neuronIndex] / targetDeltaCount[neuronIndex]
     : targetDeltaSum[neuronIndex];
   ```
   
2. **Add `normaliseGradients` config option** — Add to `BackPropagationConfig` (default: `false` for backwards compatibility, but recommended `true` for large creatures)

3. **Hybrid approach consideration** — Instead of pure averaging, consider a damped sum: `totalDelta / sqrt(targetDeltaCount)` — this preserves some gradient magnitude for highly-connected neurons while preventing extreme accumulation (similar to AdaGrad-style normalisation)

4. **Update tests** — Ensure existing convergence tests still pass. Add tests that verify normalisation improves convergence stability for networks with skewed connectivity.

### Key files to modify

- `src/propagate/TopologicalBackpropagation.ts` — lines 92, 275-276
- `src/propagate/BackPropagation.ts` — add config option
- `src/config/` — if a separate config file is needed (follow the config pattern in MEMORY.md)

### Key constraint

- Default behaviour must remain unchanged (sum, not average) to avoid breaking existing creature training
- The normalisation should be opt-in via configuration
- Must not degrade performance for small/simple creatures

## Acceptance criteria

- [ ] New config option `normaliseGradients` (or similar) added to `BackPropagationConfig`
- [ ] When enabled, gradient accumulation is normalised by downstream path count
- [ ] Default is backwards-compatible (existing behaviour unchanged)
- [ ] Tests verify improved convergence for networks with skewed connectivity
- [ ] Existing 2000+ tests continue to pass
- [ ] Australian English spelling throughout
- [ ] `./quality.sh` passes

Depends on #1870
Depends on #1871

Part of #1869

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/1889 has been created to address this issue.

**Comment by stservice** [comment]
Closed automatically — PR #1889 has been merged.

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/1890 has been created to address this issue.
